### PR TITLE
Added MessagePack parser response middleware

### PR DESCRIPTION
**Pull request by tomtaylor**
Does what it says on the tin!

### CODE REVIEW

**Comment by dasch** [comment]
@pengwynn are you still interested in this? I can rebase it on master if necessary.

**Comment by thedrow** [comment]
I'm really interested in this.

File: lib/faraday_middleware.rb [hunk 1]
```diff
@@ -1,45 +1,47 @@
...
+  autoload :EncodeJson,       'faraday_middleware/request/encode_json'
+  autoload :MethodOverride,   'faraday_middleware/request/method_override'
+  autoload :Mashify,          'faraday_middleware/response/mashify'
+  autoload :Rashify,          'faraday_middleware/response/rashify'
+  autoload :ParseJson,        'faraday_middleware/response/parse_json'
+  autoload :ParseXml,         'faraday_middleware/response/parse_xml'
+  autoload :ParseMarshal,     'faraday_middleware/response/parse_marshal'
+  autoload :ParseYaml,        'faraday_middleware/response/parse_yaml'
+  autoload :ParseDates,       'faraday_middleware/response/parse_dates'
+  autoload :ParseMessagePack, 'faraday_middleware/response/parse_dates'
```
Comment (iMacTia) [inline lib/faraday_middleware.rb]: there's a copy-paste issue here as the path is clearly wrong (loading `parse_dates` instead)

File: spec/parse_messagepack_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+require 'helper'
+require 'faraday_middleware/response/parse_message_pack'
```
Comment (iMacTia) [inline spec/parse_messagepack_spec.rb]: this require shouldn't be needed, but it is due to the issue in the autoloads above

**Comment by iMacTia** [comment]
@tomtaylor @dasch @thedrow sorry for the delay in having a look at this.
Given this looks like a useful middleware, I could merge it in, but this branch probably needs a rebase plus a couple of fixes as highlighted in my review.
Happy to have a look if you want to spend some time fixing it

**Comment by iMacTia** [comment]
Hello guys,

currently planning this for v0.11.0 release if you can merge changes from master and fix those small points in my review 😃

**Comment by iMacTia** [comment]
I had to posticipate this one guys, will add to a release when changes will be addressed

**Comment by iMacTia** [comment]
@tomtaylor @dasch @thedrow I'm preparing the next release (0.12) so if any of you want to rebase and fix the branch then this is a good time.

**Comment by thedrow** [comment]
I no longer use ruby. Never got the chance to do this.

**Comment by AlexWayfer** [comment]
I guess, due to deprecation of the whole `faraday_middleware`, it can be closed.

**Comment by iMacTia** [comment]
Good point @AlexWayfer, if anyone is interested this can be released as a standalone gem using https://github.com/lostisland/faraday-middleware-template 👍

**Pull request closed without merging**